layer_metadata_restorer_test.go

```diff
@@ -32,15 +32,18 @@ func testLayerMetadataRestorer(t *testing.T, when spec.G, it spec.S) {
 		cacheMetadata  platform.CacheMetadata
 		buildpacks     []buildpack.GroupBuildpack
 		skipLayers     bool
+		useShaFile     bool
+		logger         log.Logger
 	)
 
 	it.Before(func() {
 		var err error
 
 		layerDir, err = ioutil.TempDir("", "lifecycle-layer-dir")
 		h.AssertNil(t, err)
-		logger := &log.Logger{Handler: &discard.Handler{}}
-		layerAnalyzer = lifecycle.NewLayerMetadataRestorer(logger, layerDir, skipLayers)
+		useShaFile = true
+		logger = log.Logger{Handler: &discard.Handler{}}
```

Comment:
I created the `logger` so we can define it only once and use it when recreating `layerAnalyzer`.
Since I moved the `Removing %q, wrong sha` logic back to the `Restorer`, we can't test it here.